### DISCUSSION TITLE
Unable to run new Project on Ubuntu 24.04 - liblibSkia not found and glib_2.33 issue on glib_2.39 System

### DISCUSSION THREAD

**Discussion by MarcoGriep88** [AvaloniaUI/Avalonia#15824] [General]
Hi, i'm having some issues on running my newly generated Project in AvaloniaUI on .net 8.0 and Ubuntu 24.04.

I already installed nuget package: SkiaSharp.NativeAssets.Linux and also tried "NoDependencies". I'm not sure if the issue is in glib_2.39, because 2.33 is referenced.

## StackTrace:

/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/AvaloniaApplication1
Unhandled exception. System.TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'libSkiaSharp' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable: 
/snap/core20/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /lib/x86_64-linux-gnu/libfontconfig.so.1)
/snap/dotnet-sdk/245/shared/Microsoft.NETCore.App/8.0.5/libSkiaSharp.so: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/libSkiaSharp.so: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/runtimes/linux-x64/native/liblibSkiaSharp.so: cannot open shared object file: No such file or directory
/snap/dotnet-sdk/245/shared/Microsoft.NETCore.App/8.0.5/liblibSkiaSharp.so: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/liblibSkiaSharp.so: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/runtimes/linux-x64/native/libSkiaSharp: cannot open shared object file: No such file or directory
/snap/dotnet-sdk/245/shared/Microsoft.NETCore.App/8.0.5/libSkiaSharp: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/libSkiaSharp: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/runtimes/linux-x64/native/liblibSkiaSharp: cannot open shared object file: No such file or directory
/snap/dotnet-sdk/245/shared/Microsoft.NETCore.App/8.0.5/liblibSkiaSharp: cannot open shared object file: No such file or directory
/home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/bin/Debug/net8.0/liblibSkiaSharp: cannot open shared object file: No such file or directory

   at SkiaSharp.SkiaApi.sk_colortype_get_default_8888()
   at SkiaSharp.SKImageInfo..cctor()
   --- End of inner exception stack trace ---
   at Avalonia.Skia.PlatformRenderInterface..ctor(Nullable1 maxResourceBytes)
   at Avalonia.Skia.SkiaPlatform.Initialize(SkiaOptions options)
   at Avalonia.SkiaApplicationExtensions.<>c.<UseSkia>b__0_0()
   at Avalonia.AppBuilder.SetupUnsafe()
   at Avalonia.AppBuilder.Setup()
   at Avalonia.AppBuilder.SetupWithLifetime(IApplicationLifetime lifetime)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at AvaloniaApplication1.Program.Main(String[] args) in /home/marcogriep/RiderProjects/AvaloniaApplication1/AvaloniaApplication1/Program.cs:line 12

Process finished with exit code 134.

## csproj file

<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>WinExe</OutputType>
        <TargetFramework>net8.0</TargetFramework>
        <Nullable>enable</Nullable>
        <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
        <ApplicationManifest>app.manifest</ApplicationManifest>
        <AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="Avalonia" Version="11.0.10"/>
        <PackageReference Include="Avalonia.Desktop" Version="11.0.10"/>
        <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.10"/>
        <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.10"/>
        
        <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.10"/>
        <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="3.0.0-preview.3.1" />
    </ItemGroup>
</Project>

Hope anyone can help :)

**Comment by Gillibald** [comment]
`GLIBC_2.33' not found (required by /lib/x86_64-linux-gnu/libfontconfig.so.1)`

**Reply by MarcoGriep88** [reply]
I hoped that glib 2.39 would be downwards compatible ^

**Comment by kekekeks** [comment]
You are using .NET SDK or IDE from Snap. This is not supported and will not be supported.

Install the SDK using .deb packages, preferably from the microsoft repository and not the botched ones from ubuntu repositories.

**Reply by MarcoGriep88** [reply]
Ah okey, thanks :) I will try this.

**Reply by MarcoGriep88** [reply]
I uninstalled dotnet-sdk-8 snap package and installed debian package version. Works like a charm. Thanks alot.

**Reply by kiwidoggie** [reply]
Same thing worked for me, just install using `sudo apt install dotnet-sdk-8.0` the Microsoft packages are no longer being pushed for 24.04 and onward, see: https://github.com/dotnet/core/blob/main/linux.md
